### PR TITLE
Issue #1252 - added form level checking for duplicate entries in nameserver form

### LINKED ISSUE #1252: Bug on name servers page
Not sure if it's a bug, but, if I enter the two name server minimum, hit save, and then copy on name server into the 1st or 2nd field I'm able to save without getting the "minimum 2 required" error message.

_Originally posted by @gabydisarli in https://github.com/cisagov/manage.get.gov/issues/1236#issuecomment-1791454411_
            

Expected behavior:
On Name server page: 

New error message that says: 

Remove duplicate entry (field-level & top of page) + forcing the 2 minimum error if they have less than 2 name servers.

### PR DESCRIPTION
**Pull request by dave-kennedy-ecs**
## Ticket

Resolves #1252 

## Changes

- On Domain nameserver edit form, added duplicate entry checking on Name server field

## Context for reviewers

Registry was removing duplicates, but registrar was allowing them to be submitted.  This tickets adds a check for duplicate entries in the nameserver field in the DNS Name servers form, and raises a field level and a form level error message ("Remove duplicate entry") when a duplicate is present.

Navigate to a domain management page.  And go to the Name servers form, under DNS.  Enter in the same value for Name server 1 and Name server 2.  Click Save.  If all other validation passes, this will raise a field level error and a form level error which reads, "Remove duplicate entry")

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the assoicated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

**Review by CocoByte — APPROVED** [review]
Looks good to me.  Verified in sandbox that everything works as expected. Approved

File: src/registrar/tests/test_views.py [hunk 1]
```diff
@@ -1219,6 +1219,8 @@ def setUp(self):
         self.app.set_user(self.user.username)
         self.client.force_login(self.user)
 
+
+class TestDomainDetail(TestDomainOverview):
```
Comment (neilmb) [inline src/registrar/tests/test_views.py:1223]: **question (non-blocking)**: Same question about duplicate name changes in lots of PRs?

File: src/registrar/tests/test_views.py [hunk 2]
```diff
@@ -1506,6 +1508,30 @@ def test_domain_nameservers_form_submit_missing_host(self):
...
+
+        Uses self.app WebTest because we need to interact with forms.
+        """
+        # initial nameservers page has one server with two ips
+        nameservers_page = self.app.get(reverse("domain-dns-nameservers", kwargs={"pk": self.domain.id}))
+        session_id = self.app.cookies[settings.SESSION_COOKIE_NAME]
+        self.app.set_cookie(settings.SESSION_COOKIE_NAME, session_id)
+        # attempt to submit the form with duplicate host names of fake.host.com
+        nameservers_page.form["form-0-ip"] = ""
+        nameservers_page.form["form-1-server"] = "fake.host.com"
```
Comment (neilmb) [inline src/registrar/tests/test_views.py:1522]: **question (non-blocking)**: Where does this magic value of `fake.host.com` come from? It looks from the code like it is already in one of these form fields. Just for ease of understanding the test, could we just fill the same server name into `form-1-server` and `form-0-server` so we don't have to know this?

File: src/registrar/utility/errors.py [hunk 3]
```diff
@@ -93,6 +95,7 @@ class NameserverError(Exception):
         NameserverErrorCodes.TOO_MANY_HOSTS: ("Too many hosts provided, you may not have more than 13 nameservers."),
         NameserverErrorCodes.MISSING_HOST: ("Name server must be provided to enter IP address."),
         NameserverErrorCodes.INVALID_HOST: ("Enter a name server in the required format, like ns1.example.com"),
+        NameserverErrorCodes.DUPLICATE_HOST: ("Remove duplicate entry"),
```
Comment (neilmb) [inline src/registrar/utility/errors.py:98]: **question (non-blocking)**: Do we have a "content design" review for language like this? @michelle-rago do you know if we have a document where these error messages get sourced from and where we document them?

**Review by neilmb — COMMENTED** [review]
I only read the unit testing code and make comments about that. j/k

LGTM

File: src/registrar/utility/errors.py (hunk 3 above)
Comment (michelle-rago) [reply src/registrar/utility/errors.py:98]: @neilmb We're reviewing error/success messages. That content is in a spreadsheet which makes it much easier to review! https://drive.google.com/drive/folders/1jnRIBcr1ZhI0BFq0i6l21X0lIOtUCIPD

File: src/registrar/tests/test_views.py (hunk 1 above)
Comment (dave-kennedy-ecs) [reply src/registrar/tests/test_views.py:1223]: TestDomainOverview tests were being run multiple times.  By creating new class, TestDomainDetail, I cut the processing time of the whole suite by maybe 40 seconds
Also, for some reason, some of the iterations of the tests were failing.  Running them only once, they pass.  Probably something missing in some SetUp or TearDown.

File: src/registrar/utility/errors.py (hunk 3 above)
Comment (dave-kennedy-ecs) [reply src/registrar/utility/errors.py:98]: @neilmb and this language was in the ticket

**Review by Katherine-Osos — APPROVED** [review]

**Pull request merged**